pkg/core/config/config.go

```diff
@@ -161,6 +165,19 @@ type Config struct {
 	SysPath string `yaml:"sysPath" default:"/sys"`
 }
 
+// remapValues is used to move deprecated config values to newer keys.
+// a warning should be logged for each migrated value
+func (c *Config) remapValues() {
+	// if environment is empty, then set it to conf.cluster for legacy behavior
+	if c.Environment == "" && c.Cluster != "" {
+		log.WithFields(log.Fields{
+			"cluster":     c.Cluster,
+			"environment": c.Environment,
+		}).Warn("Config value 'cluster' is deprecated, please use `environment` instead")
```

Comment:
Yeah let's definitely take this out and use the extra span tag config on the forwarder instead.